pytorch_lightning/trainer/trainer.py

```diff
@@ -1383,15 +1383,21 @@ def __set_ckpt_path(self, ckpt_path: Optional[str], model_provided: bool, model_
             ckpt_path = "best"
 
         if ckpt_path == "best":
-            # if user requests the best checkpoint but we don't have it, error
+            if len(self.checkpoint_callbacks) > 1:
+                raise MisconfigurationException(
+                    f'.{fn}(ckpt_path="best" is not supported with multiple `ModelCheckpoint` callbacks.'
+                    " Please pass in the exact checkpoint path."
+                )
+
```

Comment:
I'd use the path of the first checkpoint with a warning to avoid breaking compatibility.

Reply:
that'd be better. will update. thank you :)